src/xma/src/xmaapi/xmahw_hal.cpp

```diff
@@ -193,13 +193,9 @@ bool hal_configure(XmaHwCfg *hwcfg, XmaXclbinParameter *devXclbins, int32_t num_
             return false;
         }
         std::string xclbin = std::string(devXclbins[i].xclbin_name);
-        char *buffer = xma_xclbin_file_open(xclbin.c_str());
-        if (!buffer)
-        {
-            xma_logmsg(XMA_ERROR_LOG, XMAAPI_MOD, "Could not open xclbin file %s\n",
-                       xclbin.c_str());
-            return false;
-        }
+        std::vector<char> xclbin_buffer = xma_xclbin_file_open(xclbin.c_str());
```

Comment:
Since xclbin is already a std::string, I recommend staying with std::string until the very lowest level when/and-only-if a const char* is required.  Alas, you should change xma_xclbin_file_open to take 'const std::string& xclbin_name'.

Reply:
done